Release preparation

mrtamm opened the pull request:
Currently, the upstream repository does releases through the goreleaser tool. If we used the same tool and configuration, we would interrupt each other. Previously [git-tag](https://github.com/CERIT-SC/funnel-gdi/blob/master-gdi/.github/workflows/git-tag.yaml) based Docker image publishing was added. Just creating a git tag would trigger building the Docker image for that version.

However, the version printed using `funnel version` command would be `unknown` as the version number was copied there by the goreleaser tool. The changes here will incorporate the same logic to the Makefile.

In addition, Dockerfile was changed so that it would use correct go version, and would also include Docker CLI tools inside the image for the default `local` worker.

(This PR also includes `master` branch changes missing from `master-gdi`.

mrtamm commented:
I updated the PR description to include the fact that now the OIDC `introspection_endpoint` is also called on validated tokens to make sure the token has not been invalidated before expiring.

dBucik commented on server/auth_oidc.go:
```diff
@@ -0,0 +1,268 @@
+
+// JSON structure of the OIDC configuration (only some fields)
+type OidcRemoteConfig struct {
+	Issuer                string `json:"issuer"`
+	JwksURI               string `json:"jwks_uri"`
+	IntrospectionEndpoint string `json:"introspection_endpoint"`
```
You can get all three of these from a so-called `well-known` endpoint (or `discovery endpoint`) of the OpenID provider.

Check out https://login.elixir-czech.org/oidc/.well-known/openid-configuration and look for `issuer, introspection_endpoint, jwks_uri`. 

However, static config here is fine by me as well as these properties do not change that often (almost not at all, perhaps during some migrations). So just a tip for consideration

mrtamm commented:
I updated authentication test `TestBasicAuthFail()` in **tests/auth/auth_test.go**. I noticed that my code changed the status code from 403 (PermissionDenied) to 401 (Unauthenticated) when user does not provide credentials in the request. Although it's a minor API change, I find it more correct than the 403 status.

I also reviewed linter errors, and fixed one linter issue in my changes.

mrtamm committed:
HTSGET and SDA storage integration for task input data (#1)

lbeckman314 committed:
docs: Update Helm Charts link in README.md

no ci

lbeckman314 committed:
fix: update website build in `hugo.yml` (#1226)

* fix: update website build in `hugo.yml`

* fix: Update hugo.yml

* Update hugo.yml

* fix: Update hugo.yml

* fix: update hugo.yml

[no ci]

mrtamm committed:
Add docker CLI to the Funnel image for the local worker

mrtamm committed:
Improve version details in the Makefile

mrtamm committed:
Merge branch 'master-gdi' into release-preparation

KrKOo committed:
Fixing S3 directory check and Kubernetes executor waiting for pod running, env variable support (#2)

* Fix GenericS3 isDir check

* Fix kubernetes executor waiting for pod running, log streaming, add env
support

mrtamm committed:
Fix: Remove whitespace from GitCommit value
